CONTRIBUTING.md

```diff
@@ -30,3 +132,49 @@ Use culturally neutral language:
 
   What you may consider well-known may be entirely obscure and distracting to people from different backgrounds.
 
+### Licensing and attribution
+
+When opening pull requests with your own contributions, you agree to licensing your work under [CC-BY-SA 4.0][cc-by-sa].
+Before merging your work, you have to sign the [contributor agreement](cla/README.md).
+
+Having a single legal entity hold non-exclusive copyright avoids disputes and ensures the material can be put to use more effectively, e.g. by eventually publishing it as a book.
+You will still be considered co-author, as recorded by version history.
+
+When adding material by third parties, make sure it has a matching license that permits this.
+In that case, [unambiguously](#links) state source, authors, and license.
+Also [add the original author as co-author][github-multiple-authors] to the respective change, so we can track authorship through version history.
+
+Notify the authors *before* using their work.
+
+[cc-by-sa]: https://creativecommons.org/licenses/by-sa/4.0/
+[github-multiple-authors]: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors
+
+### Links
+
+Unless explicitly required to point to the latest version of an external resource, all references should be [permanent links][permalink].
+
+Many web services offer permalinks.
+Examples:
+- [GitHub URLs to specific commits][github-permalink]
+- [Wikipedia URLs to specific page versions][wikipedia-permalink]
+- [Internet Archive "Save Page Now" for persisting web pages][internet-archive-save]
+
+[permalink]: https://en.m.wikipedia.org/wiki/Permalink
+[github-permalink]: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
+[wikipedia-permalink]: https://en.m.wikipedia.org/wiki/Wikipedia:Linking_to_Wikipedia#Permanent_links_to_old_versions_of_pages
+[internet-archive-save]: https://web.archive.org/save
+
+### Markdown
+
+Write one sentence per line.
+This makes long sentences immediately visible, and makes it easier to manage changes.
+
+The rule is unambiguous and does not require tooling support to be applied easily.
+[Here is a discussion of different line wrapping styles.][markdown-wrapping-styles]
+
+Use [reference links][reference-link] to keep the plain text readable.
+Collect links at the end of each section, which are delimited by headings.
+
+[markdown-wrapping-styles]: https://web.archive.org/web/20220519121408/https://mtsknn.fi/blog/4-1-wrapping-styles-for-markdown-prose-and-code-comments/
+[reference-link]: https://github.github.com/gfm/#reference-link
```

Comment:
> IMO this is only useful if the same link is used multiple times or is very long

Most links are very long. Definitely long enough to break reading flow in the source.

> enforcing it for every link is annoying

We don't have to enforce it, and apply some discretion when reviewing. It's only a guide after all. If we wanted to strictly enforce it for some reason we could add a formatter to CI.

@SuperSandro2000 In any case, the point of the proposal is to strike a balance when improving convenience for authors, reviewers, and maintainers (or anyone who wants to change the source).

For instance, we don't have to name the references if they are not reused, and keeping them close to their use site should not impose too much of a burden on maintenance. In fact, I think it is *easier* to update links when they are on a separate line, and especially easier to inspect the resulting diff.

Does that sound more reasonable? 

```suggestion
The rule is unambiguous and does not require tooling support to be applied easily.
[Here is a discussion of different line wrapping styles.]

Use [reference links] to keep the plain text readable.
Collect links close to their references:
at the end of each paragraph (delimited by an empty lines) or section (delimited by headings).

[Here is a discussion of different line wrapping styles.]: https://web.archive.org/web/20220519121408/https://mtsknn.fi/blog/4-1-wrapping-styles-for-markdown-prose-and-code-comments/
[reference links]: https://github.github.com/gfm/#reference-link
```